AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/StateMachine/Resolvers/RefreshSession/RefreshSessionState+Resolver.swift

```diff
@@ -0,0 +1,143 @@
+                if case .fetched(let identityID,
+                                 let credentials) = isAuthorizationEvent(event) {
+                    let amplifyCredentials = AmplifyCredentials.userPoolAndIdentityPool(
+                        tokens: tokens,
+                        identityID: identityID,
+                        credentials: credentials)
+                    return .init(newState: .refreshed(amplifyCredentials))
+                }
+                let resolver = FetchAuthSessionState.Resolver()
+                let resolution = resolver.resolve(oldState: fetchSessionState, byApplying: event)
+                return .init(newState: .fetchingAuthSessionWithUserPool(resolution.newState, tokens),
+                             actions: resolution.actions)
+
+            case .refreshed:
+                return .from(oldState)
+
+            case .refreshingUnAuthAWSCredentials:
+                if case .fetchedAWSCredentials(
+                    let identityID,
+                    let credentials) = isFetchSessionEvent(event) {
+                    let amplifyCredentials = AmplifyCredentials.identityPoolOnly(
+                        identityID: identityID,
+                        credentials: credentials)
+                    return .init(newState: .refreshed(amplifyCredentials))
+                }
+
+                return .from(oldState)
+            case .refreshingAWSCredentialsWithUserPoolTokens(let tokens, _):
+                if case .fetchedAWSCredentials(
+                    let identityID,
+                    let credentials) = isFetchSessionEvent(event) {
+                    let amplifyCredentials = AmplifyCredentials.userPoolAndIdentityPool(
+                        tokens: tokens,
+                        identityID: identityID,
+                        credentials: credentials)
+                    return .init(newState: .refreshed(amplifyCredentials))
+                }
+                return .from(oldState)
+            }
+        }
+
+        private func isRefreshSessionEvent(_ event: StateMachineEvent)
+        -> RefreshSessionEvent.EventType? {
+            guard let refreshSessionEvent = (event as? RefreshSessionEvent)?.eventType else {
+                return nil
+            }
+            return refreshSessionEvent
+        }
+
+        private func isAuthorizationEvent(_ event: StateMachineEvent)
+        -> AuthorizationEvent.EventType? {
+            guard let authZEvent = (event as? AuthorizationEvent)?.eventType else {
+                return nil
+            }
+            return authZEvent
+        }
+
+        private func isFetchSessionEvent(_ event: StateMachineEvent)
+        -> FetchAuthSessionEvent.EventType? {
+            guard let refreshSessionEvent = (event as? FetchAuthSessionEvent)?.eventType else {
+                return nil
+            }
+            return refreshSessionEvent
+        }
```

Comment:
I see that we use this in multiple resolvers.. May be an extension would be nice. what do you think?